pkg/gauges/contracts/LiquidityGaugeFactory.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Clones.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Ownable.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+import "./interfaces/ILiquidityGauge.sol";
+
+contract LiquidityGaugeFactory is Authentication, Ownable {
+    IVault private immutable _vault;
+    ILiquidityGauge private _gaugeImplementation;
+
+    mapping(address => bool) private _isGaugeFromFactory;
+    mapping(address => address) private _poolGauge;
```

Comment:
Note that:

a) we'll have gauges with no associated pools (e.g. the polygon gauge)
b) multiple pools can technically exist in the same contract (if the same contract calls `registerPool()` multiple times)

Reply:
a)
My understanding on this is we'll have a separate factory for gauges for each network. The `_poolGauge` mapping is very useful for L1 pools for onchain lookups in order to join+stake.

Agreed that things are less clear cut for L2s.
 
b) 
If such a contract existed it would have to track ownership in the various pools through either separate ERC20 contracts or using ERC1155 (would have to be wrapped into multiple ERC20s anyway). We're then in a situation which this contract supports.

The only thing that wouldn't work is a pool contract which has multiple subpools and enforces that liquidity provision into all of them is atomic. This design is a bit out there and I don't think we need to worry about it.

Reply:
a) I wasn't referring to the actual L2 gauges, but rather the fact that we'll have a gauge that represents all of the Polygon mining (to then bridge tokens over). That gauge will not correspond to any single pool, but rather to the network.

b) Agreed, thanks!

Reply:
We can't have a single gauge which represents all the Polygon pools as we need the ability to vote for individual pools on Polygon. Each Polygon pool will have a corresponding gauge on mainnet.

Reply:
You're right, I was confused. Given we have address and ID collisions, I propose then that we use an augmented pool ID that includes the network as the identifier, instead of the plain address.

Reply:
@TomAFrench should we augment pool addresses with chain ID? If so, would we do this in this PR?

Reply:
I don't think so, this contract will only be deploying the gauges that will be used for mainnet pools. We'd need a separate factory for Polygon pools, and another for Arbitrum. At that point the factory address is a good disambiguator for which network the gauge is for.